### PR TITLE
feat(logs): sanitize basic auth in urls

### PR DESCRIPTION
**Pull request by danwilldev**
This regex is not perfect, any @ or : characters in a password will cause it to fail. If someone could suggest a better way of handling it then please comment or commit to this PR.
Thanks

### CODE REVIEW

**Comment by danwilldev** [comment]
Test failing. Think I compiled the regex incorrectly

**Comment by KyleSanderson** [comment]
might be worthwhile while specifying REDACTED_USER/etc. I don't think we do that elsewhere but it helps add clarity than just REDACTED literred everywhere

File: internal/http/logs.go [hunk 1]
```diff
@@ -139,6 +139,10 @@ var (
 		{
 			pattern: regexp.MustCompile(`(SceneHD..invite).([a-zA-Z0-9]+)(\s+#announce)`),
 			repl:    "$1 REDACTED$3",
+		},
+				{
+			pattern: regexp.MustCompile(`(https?://)(.*?):(.*?)@(.*?)`),
```
Comment (KyleSanderson) [inline internal/http/logs.go:144]: the `:(.*?)` of this is optional, and won't match if the password is absent.

**Comment by zze0s** [comment]
Looks like this needs some more love before it's ready to merge.

**Comment by zze0s** [comment]
@danwilldev I was playing with this to try get it passing tests, but could not figure out why it does not work as expected. Maybe marty or soup can look into it since they built the log sanitizer.

**Comment by danwilldev** [comment]
@zze0s it may be because I'm using "username:password" in the test string so it's getting double sanitized? Will leave it for now and hope one of the regex pros picks it up :D

**Review by s0up4200 — APPROVED** [review]

**Pull request merged**